src/viam/sdk/common/utils.hpp

```diff
@@ -58,13 +58,32 @@ class ClientContext {
     ClientContext();
     operator grpc::ClientContext*();
     operator const grpc::ClientContext*() const;
+    void set_debug_key(const std::string& debug_key);
 
    private:
     void set_client_ctx_authority_();
     void add_viam_client_version_();
     grpc::ClientContext wrapped_context_;
 };
 
+/// @brief Returns a new `AttributeMap` with a random key for server-side debug logging
+AttributeMap debug_map();
+
+/// @brief Returns a new `AttributeMap` with @param debug_key for server-side debug logging
+/// @throws Exception if the debug_key contains invalid (e.g., uppercase) gRPC characters
+AttributeMap debug_map(std::string debug_key);
+
+/// @brief Adds @param debug_key for server-side debug logging to @param map
+/// @throws Exception if the debug_key contains invalid (e.g., uppercase )gRPC characters
+AttributeMap add_debug_entry(AttributeMap&& map, std::string debug_key);
+
+/// @brief Adds @param debug_key for server-side debug logging to @param map
+/// @throws Exception if the debug_key contains invalid (e.g., uppercase )gRPC characters
+AttributeMap add_debug_entry(AttributeMap map, std::string debug_key);
+
+/// @brief Adds a random key to @param map for server-side debug logging
+AttributeMap add_debug_entry(AttributeMap&& map);
```

Comment:
This and the one above should probably pass the parameter in the same way?

Reply:
Hmm maybe you can help me out here! I have an `add_debug_entry` that passes the param the same way above, but with just those two I get compilation errors (`candidate function not viable: expects an rvalue for 1st argument`). Having the "pass by value" version fixed that, though perhaps it shouldn't be exposed to users?